core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/UserApiService.scala

```diff
@@ -23,126 +23,147 @@ trait UserApiService extends HttpService with PerRequestCreator with UserInfoDir
 
   // these routes have different access control and their paths have /register instead of /api
 
-  val createUserRoute = requireUserInfo() { userInfo =>
-    path("user") {
-      post {
-        requestContext => perRequest(requestContext,
-          UserService.props(userServiceConstructor, userInfo),
-          UserService.CreateUser)
+  val createUserRoute = detach() {
+    requireUserInfo() { userInfo =>
+      path("user") {
+        detach() {
+          post {
+            requestContext =>
+              perRequest(requestContext,
+                UserService.props(userServiceConstructor, userInfo),
+                UserService.CreateUser)
```

Comment:
You should probably note here that we need to switcheroo here not because pets will ever be in a situation where they'll _need_ to create their owners, but that this endpoint needs to behave correctly if a pet _does_ call this endpoint. If you didn't do a switcheroo here the pet would get created as a new user of its own.